### PR TITLE
[Fix](inverted index) fix memory leak of inverted index writer for array values

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

Issue Number: close #xxx

fix inverted index writer's field leak

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp [hunk 1]
```diff
@@ -385,13 +385,6 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
                     if (null_map[j] == 1) {
                         continue;
                     }
-                    // now we temp create field . later make a pool
-                    if (Status st = create_field(&new_field); st != Status::OK()) {
```
Comment (amorynan) [inline be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:389]: here to create field would make field memory leak?

**Comment by airborne12** [comment]
run buildall

**Review by amorynan — APPROVED** [review]
LGTM

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**